### PR TITLE
External documentation website: Searchbox feature

### PR DESCRIPTION
**Pull request by sopranopillow**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

Added a searchbox to the left nav that filters the components in the list.
First photo is the original and the other two have the searchbox.

original look
searchbox text and clear button
searchbox

#### Focus areas to test

(optional)

### CODE REVIEW

File: apps/fabric-website/src/components/Nav/Nav.tsx [hunk 1]
```diff
@@ -53,23 +66,74 @@ export class Nav extends React.Component<INavProps, INavState> {
...
+            }}
+            onChange={this._onChangeQuery.bind(this)}
+          />
+        </div>
+      );
+    }
+  }
+
+   private _renderLinkList(pages: INavPage[], isSubMenu: boolean): React.ReactElement<{}> {
+    let { searchQuery } = this.state;
```
Comment (kenotron) [inline apps/fabric-website/src/components/Nav/Nav.tsx]: let -> const

File: apps/fabric-website/src/components/Nav/Nav.tsx [hunk 2]
```diff
@@ -53,23 +66,74 @@ export class Nav extends React.Component<INavProps, INavState> {
...
-
-    return (
-      <ul className={css(styles.links, isSubMenu ? styles.isSubMenu : '')} aria-label="Main website navigation">
-        {links}
-      </ul>
-    );
+      .filter(
+        page =>
+          page.title.toLowerCase().indexOf(searchQuery.toLowerCase()) > -1 ||
+          page.url.toLowerCase().indexOf('#/components/')
```
Comment (kenotron) [inline apps/fabric-website/src/components/Nav/Nav.tsx]: indexOf(...) > -1, right?

Comment (kenotron) [reply apps/fabric-website/src/components/Nav/Nav.tsx]: or do you really mean indexOf() == 0 ?

Comment (sopranopillow) [reply apps/fabric-website/src/components/Nav/Nav.tsx]: it only works like that, ==0 and > -1 hides links that are not part of the components list

Comment (kenotron) [reply apps/fabric-website/src/components/Nav/Nav.tsx]: Then it should be >0

**Pull request merged**